pyvista/core/filters/data_set.py

```diff
@@ -47,8 +48,10 @@
 class DataSetFilters:
     """A set of common filters that can be applied to any vtkDataSet."""
 
-    def _clip_with_function(
-        self,
+    points: pyvista.pyvista_ndarray
```

Comment:
This line is needed to resolve a bunch of type errors caused by #5434.

If we remove this line, it generates errors like:
``` python
pyvista/core/filters/poly_data.py:2838: error: Cannot determine type of "points"  [has-type]
            np.apply_along_axis(f, 1, mesh.points)
```
``` python
pyvista/core/dataset.py:439: error: Signature of "points" incompatible with supertype "DataSetFilters"  [override]
        @property
         ^
pyvista/core/dataset.py:439: note:      Superclass:
pyvista/core/dataset.py:439: note:          ndarray[Any, dtype[floating[_32Bit]]]
pyvista/core/dataset.py:439: note:      Subclass:
pyvista/core/dataset.py:439: note:          pyvista_ndarray
```

It is a bit odd to have this type hint here though. Alternatively, we can remove it and simply `# type: ignore` the individual errors on a per-line basis.

Reply:
The hint was removed in [d5ab232](https://github.com/pyvista/pyvista/pull/6870/commits/d5ab232c147dad2696251ad422fb2cf012a8fc3f) and type ignores added instead. This is better, because if https://github.com/pyvista/pyvista/issues/5434 ever gets fixed, then mypy should auto-detect that the ignores are no longer needed at that time. By contrast, if we keep the property's type hint we won't get notified by mypy that it's redundant (and we won't know to remove it later).